docs/website/docs/building-from-source/riscv.md

```diff
@@ -0,0 +1,137 @@
+# RISC-V cross-compilation
+
+Running on a platform like RISC-V involves cross-compiling from a _host_
+platform (e.g. Linux) to a _target_ platform (a specific RISC-V CPU architecture
+and operating system):
+
+* IREE's _compiler_ is built on the host and is used there to generate modules
+  for the target
+* IREE's _runtime_ is built on the host for the target. The runtime is then
+  pushed to the target to run natively.
+
+## Prerequisites
+
+### Host environment setup
+
+You should already be able to build IREE from source on your host platform.
+Please make sure you have followed the [getting started](./getting-started.md)
+steps.
+
+### Install RISC-V cross-compile toolchain and emulator
+
+You'll need a RISC-V LLVM compilation toolchain and a RISC-V enabled QEMU
+emulator.
+
+* RISC-V toolchain is built from https://github.com/llvm/llvm-project (main branch).<br>
+  * Currently, the LLVM compiler is built on GNU toolchain, including libgcc,
+    GNU linker, and C libraries. You need to build GNU toolchain first.<br>
+  * Clone GNU toolchain from: https://github.com/riscv/riscv-gnu-toolchain
+    (master branch). Switch the "riscv-binutils" submodule to `rvv-1.0.x-zfh`
```

Comment:
This instruction is very specific. Can we simplify this or link to an external guide instead?

Reply:
This is tricky as the vector extension instructions are not enabled in the master branch so to use the vector extension ops a specific `riscv-binutils` branch is used.

There's no public doc of the RVV support we can refer to yet. I can either add more explanation w.r.t the vector extension or not supporting the vector extension in this doc.

Reply:
Per offline discussion. Only provide the external links for the generic RISC-V toolchain support. Leave the `v` CPU feature out of this doc.

Reply:
Another idea is to add a section at the bottom of the document for configuration options, possibly with a link from higher up. The main path through the document should be simple to follow, then dedicated users can go through the additional sections.

rough outline:
```
### Install RISC-V cross-compile toolchain and emulator

do X, Y, and Z
Note: for vector extensions support, see [link to header down in the page].

...

## Optional configuration

Vector extensions let SIMD code run more efficiently ... 

check out the `foo` branch on this repo and ...
```

Reply:
Optional section added at the end of the doc.